archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,754 @@
+import datetime
+import errno
+import json
+import logging
+import os
+import re
+import subprocess
+import shutil
+import tarfile
+import uuid
+
+from arteria.exceptions import ArteriaUsageException
+from arteria.web.state import State
+from arteria.web.handlers import BaseRestHandler
+
+from archive_upload import __version__ as version
+from archive_upload.lib.jobrunner import LocalQAdapter
+
+log = logging.getLogger(__name__)
+
+class BaseDsmcHandler(BaseRestHandler):
+    """
+    Base handler for dsmc upload operations.
+    """
+
+    def initialize(self, config, runner_service):
+        """
+        Ensures that any parameters feed to this are available
+        to subclasses.
+
+        :param config: configuration used by the service
+        :param runner_service: runner service to use. Must fulfill `archive_upload.lib.jobrunner.JobRunnerAdapter` interface
+        """
+        self.config = config
+        self.runner_service = runner_service
+
+    @staticmethod
+    def _validate_runfolder_exists(runfolder, monitored_dir):
+        """
+        Validate that the runfolder exists under monitored directories
+        :param runfolder: The runfolder to check for
+        :param monitored_dir: The root in which the runfolder should exist
+        :return: True if this is a valid runfolder
+        """
+
+        if os.path.isdir(monitored_dir):
+            sub_folders = [ name for name in os.listdir(monitored_dir)
+                            if os.path.isdir(os.path.join(monitored_dir, name)) ]
+            return runfolder in sub_folders
+        else:
+            return False
+
+    @staticmethod
+    def _is_valid_log_dir(log_dir):
+        """
+        Check if the log dir is valid. Right now only checks it is a directory.
+        :param: log_dir to check
+        :return: True is valid dir, else False
+        """
+        return os.path.isdir(log_dir)
+
+    @staticmethod
+    def _rm_empty_dirs(path, remove_root=True):
```

Comment:
My IDE warns that there are multiple lines in this method where the indentation is not a multiple of 4.

Reply:
Additionally the logic here could probably be replaced with:
```
import shutil
shutil.rmtree('/path/to/your/dir/')
```

Reply:
No, I don't want to remove everything. Only empty directories. (It is used for cleaning up the archive after we have created the compressed file.)

Reply:
Aha. Then I get it.